optimizely/helpers/condition.py

```diff
@@ -105,10 +146,31 @@ def greater_than_evaluator(self, index):
           - False if the user attribute value is less than or equal to the condition value.
         None: if the condition value isn't finite or the user attribute value isn't finite.
     """
+    condition_name = self.condition_data[index][0]
     condition_value = self.condition_data[index][1]
-    user_value = self.attributes.get(self.condition_data[index][0])
+    user_value = self.attributes.get(condition_name)
+
+    if isinstance(user_value, (numbers.Integral, float)) and \
+            abs(user_value) > (2**53):
+      self.logger.debug(logs.INFINITE_ATTRIBUTE_VALUE.format(
+        self._get_condition_log(index),
+        user_value
+      ))
+      return None
```

Comment:
I think it's too subtle to merely perform our numeric bounds check before the type check.

It'd be cleaner to restrict the type check such that it doesn't do _any_ bounds checking (renaming it from `is_finite_number ` to `is_number `) and then perform the numeric bounds check afterward. That's what we settled on [in the javascript-sdk](https://github.com/optimizely/javascript-sdk/blob/78ba995457e202ebb34f5dc0985255872e726c29/packages/optimizely-sdk/lib/core/custom_attribute_condition_evaluator/index.js#L171-L179).

Then we'd be able to order our checks from most general to most specific, per convention.